ccxt.d.ts

```diff
@@ -1,189 +1,430 @@
 declare module 'ccxt' {
 
-    export interface Market {
-        id: string;
-        symbol: string;
+    // error.js -----------------------------------------
+    export class BaseError extends Error {
+        constructor(message: string);
+    }
+
+    export class ExchangeError extends BaseError {
+        constructor(message: string);
+    }
+
+    export class NotSupported extends ExchangeError {
+        constructor(message: string);
+    }
+
+    export class AuthenticationError extends ExchangeError {
+        constructor(message: string);
+    }
+
+    export class InvalidNonce extends ExchangeError {
+        constructor(message: string);
+    }
+
+    export class InsufficientFunds extends ExchangeError {
+        constructor(message: string);
+    }
+
+    export class InvalidOrder extends ExchangeError {
+        constructor(message: string);
+    }
+
+    export class OrderNotFound extends InvalidOrder {
+        constructor(message: string);
+    }
+
+    export class OrderNotCached extends InvalidOrder {
+        constructor(message: string);
+    }
+
+    export class CancelPending extends InvalidOrder {
+        constructor(message: string);
+    }
+
+    export class NetworkError extends BaseError {
+        constructor(message: string);
+    }
+
+    export class DDoSProtection extends NetworkError {
+        constructor(message: string);
+    }
+
+    export class RequestTimeout extends NetworkError {
+        constructor(message: string);
+    }
+
+    export class ExchangeNotAvailable extends NetworkError {
+        constructor(message: string);
+    }
+
+    // -----------------------------------------------
+
+    export interface IMinMax {
+        max: number;
+        min: number;
+    }
+
+    export interface IMarket {
+        [key: string]: any
         base: string;
-        quote: string;
+        id: string;
         info: any;
+        limits: { amount: IMinMax, price: IMinMax, cost?: IMinMax };
+        precision: { amount: number, price: number };
+        quote: string;
+        symbol: string;
     }
 
-    export interface OrderBook {
-        bids: number[][];
-        asks: number[][];
-        timestamp: number;
+    export interface IMarketInfo {
```

Comment:
Do you think we need this interface? I can't find any reference to it anywhere. I guess, this is something you use in your project?

Reply:
We need this for `getMarket` method . This one returns an instance of `Market` class. There was an error with the return type of this method which is already fixed. I think I'm going to change names between `IMarket` and `IMarketInfo` interfaces because `IMarketInfo` represents the `Market` class.